stock_valuation_fifo_lot/models/stock_valuation_layer.py

```diff
@@ -12,6 +14,30 @@ class StockValuationLayer(models.Model):
         string="Lots/Serials",
     )
 
+    def write(self, vals):
+        """Override the remaining value according to the consumed value of the
+        corresponding stock move line for lot-managed FIFO products.
+        """
+        ctx = self.env.context
+        fifo_lot = ctx.get("fifo_lot")
+        consumed = ctx.get("consumed_value")
+        # i.e., the layer is a candidate layer for lot managed FIFO product
+        if fifo_lot and consumed is not None and "remaining_value" in vals:
+            self.ensure_one()
+            remaining_value = self.remaining_value - ctx.get("consumed_value")
+            if (
+                float_compare(
+                    remaining_value, 0.0, precision_rounding=self.currency_id.rounding
+                )
+                < 0
+            ):
+                raise ValidationError(
+                    _("Remaining Value cannot be negative for the candidate layer.")
+                )
```

Comment:
This will block the operation if we are allowed to receive with existing lot and deliver at once.

- Receive 1 quantity with lot 001 for receipt 1
- Receive 1 quantity with lot 001 for receipt 2
- Deliver 2 quantity with lot 001

The consumed_value is always greater than remaining value of each SVL.
We may just check `value_remain_lot` and `consumed_value` before passing context.
https://github.com/qrtl/axls-oca/pull/186/files#diff-c963c501ed0f753745eb335c6ac86ea98eb6a7661e08198fe1924822d58796aaR100-R103

Reply:
@AungKoKoLin1997 Thanks for this. That's a very valid point. Can you please check my last commit?

Reply:
@yostashiro 
I am not sure about below scenario can happen in real business but technically it can happen and there will be an issue.

- Receive 1 quantity with lot 001 for receipt 1 and price unit is 100
- Receive 1 quantity with lot 001 for receipt 2 and price unit is 200
- Deliver 2 quantity with lot 001

As per current design, the unit_cost is 150 and compare with remaining_value of each SVL.
So, SVL with remaining_value 100 will block the operation.